front/src/routes/dashboard/edit-dashboard/EditBoxColumns.jsx

```diff
@@ -8,145 +9,233 @@ import EditBox from './EditBox';
 import EmptyColumnDropZone from './EmptyColumnDropZone';
 import BottomDropZone from './BottomDropZone';
 import AutoScrollMobile from '../../../components/drag-and-drop/AutoScrollMobile';
-import style from '../style.css';
+import style from './style.css';
+import stylePrimary from '../style.css';
 import { DASHBOARD_VISIBILITY_LIST } from '../../../../../server/utils/constants';
 
 const DASHBOARD_EDIT_BOX_TYPE = 'DASHBOARD_EDIT_BOX';
 
-const EditBoxColumns = ({ children, ...props }) => (
-  <div class="pb-6">
-    <h3>
-      <Text id="dashboard.editDashboardTitle" />
-    </h3>
-    {props.dashboardAlreadyExistError && (
-      <div class="alert alert-danger">
-        <Text id="newDashboard.dashboardAlreadyExist" />
-      </div>
-    )}{' '}
-    {props.dashboardValidationError && (
-      <div class="alert alert-danger">
-        <Text id="newDashboard.validationError" />
-      </div>
-    )}
-    {props.unknownError && (
-      <div class="alert alert-danger">
-        <Text id="newDashboard.unknownError" />
-      </div>
-    )}
-    <div class="row align-items-end">
-      <div class="col-md-4">
-        <div class="form-group">
-          <label class="form-label">
-            <Text id="dashboard.editDashboardNameLabel" />
-          </label>
-          <Localizer>
-            <input
-              type="text"
-              class="form-control"
-              placeholder={<Text id="dashboard.editDashboardNameLabel" />}
-              value={props.homeDashboard.name}
-              onInput={props.updateCurrentDashboardName}
-            />
-          </Localizer>
+const EditBoxColumns = ({ children, ...props }) => {
+  const columnCardRef = useRef(null);
+  const maxBoxes = 3;
+  const lengthBoxes = props.homeDashboard.boxes.length;
+  const [parentHeight, setParentHeight] = useState(0);
+
+  useEffect(() => {
```

Comment:
Is the use of hooks absolutely needed here?

I'm not in favor of hooks in Gladys, those leads to nasty bugs and most people doesn't understand them.

We do use hooks in one or two places in Gladys, but because we use a third party library that requires it.

Can you switch to a class component here?